File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(nums) {
        let l = 0
        let r = nums.length - 1;
        let minVal = 1001
        while(l < r){
            let mid = Math.floor((l + r) / 2)
            minVal = Math.min(minVal, nums[mid])
            if(nums[mid] > nums[r]){
                l = mid + 1
            }else if(nums[mid] < nums[l]){
                r = mid - 1
            }else{
                break
            }
        }
        return Math.min(minVal, nums[l])
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        beg = 0
        for i in range(len(nums)):
            if(i > 0 and nums[i] != nums[i - 1]):
                nums[beg + 1] = nums[i]
                beg += 1

        return beg + 1